astropy/visualization/wcsaxes/axislabels.py

```diff
@@ -1,16 +1,17 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+import warnings
 
 import matplotlib.transforms as mtransforms
 import numpy as np
-from matplotlib import rcParams
+from matplotlib import _api, rcParams
 from matplotlib.text import Text
 
 from .frame import RectangularFrame
 
 
 class AxisLabels(Text):
-    def __init__(self, frame, minpad=1, *args, **kwargs):
+    def __init__(self, frame, minpad=1, loc="center", *args, **kwargs):
```

Comment:
This argument should be keyword only. If only to mimic matplotlib's API
```suggestion
    def __init__(self, frame, minpad=1, *args,  loc="center", **kwargs):
```
It'll also avoid breaking backward compatibility with anyone passing positional args currently captured in `*args`